src/main/java/hudson/tasks/junit/TestResult.java

```diff
@@ -129,27 +141,37 @@ public TestResult(boolean keepLongStdio) {
     }
 
     @Deprecated
-    public TestResult(long buildTime, DirectoryScanner results) throws IOException {
-        this(buildTime, results, false);
+    public TestResult(long filesTimestamp, DirectoryScanner results) throws IOException {
+        this(filesTimestamp, results, false);
     }
 
     @Deprecated
-    public TestResult(long buildTime, DirectoryScanner results, boolean keepLongStdio) throws IOException {
-        this(buildTime, results, keepLongStdio, null);
+    public TestResult(long filesTimestamp, DirectoryScanner results, boolean keepLongStdio) throws IOException {
+        this(filesTimestamp, results, keepLongStdio, null);
     }
 
+    @Deprecated
+    public TestResult(long filesTimestamp, DirectoryScanner results, boolean keepLongStdio,
+                      PipelineTestDetails pipelineTestDetails) throws IOException {
+        this.keepLongStdio = keepLongStdio;
+        impl = null;
+        parse(filesTimestamp, results, pipelineTestDetails);
```

Comment:
Should probably be made into a `this`-constructor call and `parseOldReports` made `final` here?